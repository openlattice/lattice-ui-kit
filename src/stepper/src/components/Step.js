// @flow

import React from 'react';
import type { ChildrenArray } from 'react';

import StepIcon from './StepIcon';
import { StepWrapper } from './styled';

type Props = {
  index :number;
  active :boolean;
  complete :boolean;
  children :ChildrenArray<any>;
  onClick ? :(e :SyntheticEvent<HTMLDivElement>) => void;
};

const Step = ({
  active,
  complete,
  children,
  index,
  onClick
} :Props) => (
  <StepWrapper onClick={onClick}>
    <StepIcon
        index={index}
        active={active}
        complete={complete} />
    {children}
  </StepWrapper>
);

Step.defaultProps = {
  onClick: undefined
};

export default Step;
