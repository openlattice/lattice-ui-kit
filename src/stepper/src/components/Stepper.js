// @flow
import React from 'react';
import type { ChildrenArray } from 'react';

import { StepDivider, StepperWrapper } from './styled';

type Props = {
  activeStep :number;
  children :ChildrenArray<any>;
};

const Stepper = ({ activeStep = 0, children } :Props) => {
  const steps = React.Children.toArray(children).map((child, index) => {

    const state = {
      active: activeStep === index,
      complete: activeStep > index,
      disabled: activeStep < index,
      index,
    };

    return [
      index !== 0 && <StepDivider />,
      React.cloneElement(child, state)
    ];
  });

  return (
    <StepperWrapper>
      {steps}
    </StepperWrapper>
  );
};

export default Stepper;
