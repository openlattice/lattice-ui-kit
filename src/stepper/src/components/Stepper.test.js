import toJson from 'enzyme-to-json';
import { mount, shallow } from 'enzyme';

import Step from './Step';
import Stepper from './Stepper';
import {
  StepDivider,
  StepperWrapper,
} from './styled';

describe('Stepper', () => {
  test('should match snapshot', () => {
    const wrapper = mount(
      <Stepper>
        <Step>Step 1</Step>
        <Step>Step 2</Step>
        <Step>Step 3</Step>
      </Stepper>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('orientation', () => {

    test('`vertical = false` should apply vertical styles', () => {
      const wrapper = mount(<Stepper />);
      const stepperWrapper = wrapper.find(StepperWrapper);
      expect(stepperWrapper.prop('vertical')).toBe(false);
      expect(stepperWrapper).toHaveStyleRule('align-items', 'center');
    });

    test('`vertical = true` should apply vertical styles', () => {
      const wrapper = mount(<Stepper vertical />);
      const stepperWrapper = wrapper.find(StepperWrapper);
      expect(stepperWrapper.prop('vertical')).toBe(true);
      expect(stepperWrapper).toHaveStyleRule('align-items', 'flex-start');
      expect(stepperWrapper).toHaveStyleRule('flex-direction', 'column');
      expect(stepperWrapper).toHaveStyleRule('justify-content', 'start');
    });
  });

  describe('children', () => {
    test('should add dividers between each child', () => {
      const wrapper = shallow(
        <Stepper>
          <div />
          <div />
          <div />
        </Stepper>
      );

      const children = wrapper.children();
      expect(children).toHaveLength(5);
      expect(wrapper.childAt(1).find(StepDivider)).toHaveLength(1);
      expect(wrapper.childAt(3).find(StepDivider)).toHaveLength(1);
    });

    test('pass orientation to children', () => {
      const wrapper = mount(
        <Stepper vertical>
          <Step />
          <Step />
          <Step />
        </Stepper>
      );
      const children = wrapper.children();
      children.forEach((child) => {
        expect(child.prop('vertical')).toBe(true);
      });

    });

    test('each step gets an index', () => {
      const wrapper = mount(
        <Stepper>
          <Step />
          <Step />
          <Step />
        </Stepper>
      );

      const steps = wrapper.find(Step);
      steps.forEach((step, index) => {
        expect(step.prop('index')).toBe(index + 1);
      });
    });

    test('`sequential` should influence state by `activeStep`', () => {
      const wrapper = shallow(
        <Stepper sequential activeStep={0}>
          <Step />
          <Step />
          <Step />
        </Stepper>
      );

      const steps = wrapper.find(Step);
      // Step 1
      expect(steps.at(0).prop('active')).toBe(true);
      expect(steps.at(0).prop('complete')).toBe(false);
      expect(steps.at(0).prop('disabled')).toBe(false);
      // Step 2
      expect(steps.at(1).prop('active')).toBe(false);
      expect(steps.at(1).prop('complete')).toBe(false);
      expect(steps.at(1).prop('disabled')).toBe(true);
      // Step 3
      expect(steps.at(2).prop('active')).toBe(false);
      expect(steps.at(2).prop('complete')).toBe(false);
      expect(steps.at(2).prop('disabled')).toBe(true);

      wrapper.setProps({ activeStep: 1 });

      const newSteps = wrapper.find(Step);

      // Step 1
      expect(newSteps.at(0).prop('active')).toBe(false);
      expect(newSteps.at(0).prop('complete')).toBe(true);
      expect(newSteps.at(0).prop('disabled')).toBe(false);
      // Step 2
      expect(newSteps.at(1).prop('active')).toBe(true);
      expect(newSteps.at(1).prop('complete')).toBe(false);
      expect(newSteps.at(1).prop('disabled')).toBe(false);
      // Step 3
      expect(newSteps.at(2).prop('active')).toBe(false);
      expect(newSteps.at(2).prop('complete')).toBe(false);
      expect(newSteps.at(2).prop('disabled')).toBe(true);

    });

    test('`sequential = false` should not influence state by `activeStep`', () => {
      const wrapper = shallow(
        <Stepper activeStep={0}>
          <Step />
          <Step />
          <Step />
        </Stepper>
      );

      const steps = wrapper.find(Step);
      // Step 1
      expect(steps.at(0).prop('active')).toBe(true);
      expect(steps.at(0).prop('complete')).toBe(false);
      expect(steps.at(0).prop('disabled')).toBe(false);
      // Step 2
      expect(steps.at(1).prop('active')).toBe(false);
      expect(steps.at(1).prop('complete')).toBe(false);
      expect(steps.at(1).prop('disabled')).toBe(false);
      // Step 3
      expect(steps.at(2).prop('active')).toBe(false);
      expect(steps.at(2).prop('complete')).toBe(false);
      expect(steps.at(2).prop('disabled')).toBe(false);

      wrapper.setProps({ activeStep: 1 });

      const newSteps = wrapper.find(Step);

      // Step 1
      expect(newSteps.at(0).prop('active')).toBe(false);
      expect(newSteps.at(0).prop('complete')).toBe(false);
      expect(newSteps.at(0).prop('disabled')).toBe(false);
      // Step 2
      expect(newSteps.at(1).prop('active')).toBe(true);
      expect(newSteps.at(1).prop('complete')).toBe(false);
      expect(newSteps.at(1).prop('disabled')).toBe(false);
      // Step 3
      expect(newSteps.at(2).prop('active')).toBe(false);
      expect(newSteps.at(2).prop('complete')).toBe(false);
      expect(newSteps.at(2).prop('disabled')).toBe(false);

    });

    test('compose props passed from children', () => {
      const wrapper = mount(
        <Stepper>
          <Step foo="bar" complete />
        </Stepper>
      );

      const step = wrapper.find(Step);
      expect(step.prop('foo')).toBe('bar');
      expect(step.prop('complete')).toBe(true);
    });
  });
});
