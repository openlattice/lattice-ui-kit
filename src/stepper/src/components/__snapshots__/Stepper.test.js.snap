// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper should match snapshot 1`] = `
.c2 {
  font-size: 24px;
  min-width: 24px;
}

.c5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  border-top: 2px solid #DFDFE3;
}

.c3 {
  font-size: 14px;
  font-weight: bold;
}

.c4 {
  margin-left: 8px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<Stepper
  sequential={false}
  vertical={false}
>
  <styled__StepperWrapper
    vertical={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__StepperWrapper-g8vnjr-5",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "align-items:center;display:flex;width:100%;",
              [Function],
              ";",
            ],
          },
          "displayName": "styled__StepperWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__StepperWrapper-g8vnjr-5",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      vertical={false}
    >
      <div
        className="c0"
      >
        <Step
          active={false}
          complete={false}
          disabled={false}
          index={1}
          key=".0/.1"
        >
          <styled__StepWrapper>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styled__StepWrapper-g8vnjr-4",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "display:flex;align-items:center;padding:10px;cursor:",
                      [Function],
                      ";",
                    ],
                  },
                  "displayName": "styled__StepWrapper",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styled__StepWrapper-g8vnjr-4",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                <StepIcon
                  active={false}
                  complete={false}
                  index={1}
                >
                  <styled__IconLayer
                    className="fa-layers"
                  >
                    <StyledComponent
                      className="fa-layers"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "styled__IconLayer-g8vnjr-0",
                            "isStatic": false,
                            "lastClassName": "c2",
                            "rules": Array [
                              "font-size:24px;min-width:24px;",
                            ],
                          },
                          "displayName": "styled__IconLayer",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "styled__IconLayer-g8vnjr-0",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <span
                        className="c2 fa-layers"
                      >
                        <FontAwesomeIcon
                          border={false}
                          className=""
                          color="#8B8B90"
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                512,
                                512,
                                Array [],
                                "f111",
                                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z",
                              ],
                              "iconName": "circle",
                              "prefix": "fas",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          swapOpacity={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-circle fa-w-16 "
                            color="#8B8B90"
                            data-icon="circle"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 512 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                        <styled__StepIndex
                          className="fa-layers-text fa-inverse"
                        >
                          <StyledComponent
                            className="fa-layers-text fa-inverse"
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "styled__StepIndex-g8vnjr-2",
                                  "isStatic": false,
                                  "lastClassName": "c3",
                                  "rules": Array [
                                    "font-size:14px;font-weight:bold;",
                                  ],
                                },
                                "displayName": "styled__StepIndex",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "styled__StepIndex-g8vnjr-2",
                                "target": "span",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <span
                              className="c3 fa-layers-text fa-inverse"
                            >
                              1
                            </span>
                          </StyledComponent>
                        </styled__StepIndex>
                      </span>
                    </StyledComponent>
                  </styled__IconLayer>
                </StepIcon>
                <styled__StepLabel
                  active={false}
                >
                  <StyledComponent
                    active={false}
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "styled__StepLabel-g8vnjr-3",
                          "isStatic": false,
                          "lastClassName": "c4",
                          "rules": Array [
                            "margin-left:8px;font-weight:",
                            [Function],
                            ";",
                          ],
                        },
                        "displayName": "styled__StepLabel",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "styled__StepLabel-g8vnjr-3",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="c4"
                    >
                      Step 1
                    </span>
                  </StyledComponent>
                </styled__StepLabel>
              </div>
            </StyledComponent>
          </styled__StepWrapper>
        </Step>
        <styled__StepDivider
          key=".1/.0"
          vertical={false}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styled__StepDivider-g8vnjr-1",
                  "isStatic": false,
                  "lastClassName": "c5",
                  "rules": Array [
                    "flex:1 1 auto;",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "styled__StepDivider",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styled__StepDivider-g8vnjr-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            vertical={false}
          >
            <div
              className="c5"
            />
          </StyledComponent>
        </styled__StepDivider>
        <Step
          active={false}
          complete={false}
          disabled={false}
          index={2}
          key=".1/.1"
        >
          <styled__StepWrapper>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styled__StepWrapper-g8vnjr-4",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "display:flex;align-items:center;padding:10px;cursor:",
                      [Function],
                      ";",
                    ],
                  },
                  "displayName": "styled__StepWrapper",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styled__StepWrapper-g8vnjr-4",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                <StepIcon
                  active={false}
                  complete={false}
                  index={2}
                >
                  <styled__IconLayer
                    className="fa-layers"
                  >
                    <StyledComponent
                      className="fa-layers"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "styled__IconLayer-g8vnjr-0",
                            "isStatic": false,
                            "lastClassName": "c2",
                            "rules": Array [
                              "font-size:24px;min-width:24px;",
                            ],
                          },
                          "displayName": "styled__IconLayer",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "styled__IconLayer-g8vnjr-0",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <span
                        className="c2 fa-layers"
                      >
                        <FontAwesomeIcon
                          border={false}
                          className=""
                          color="#8B8B90"
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                512,
                                512,
                                Array [],
                                "f111",
                                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z",
                              ],
                              "iconName": "circle",
                              "prefix": "fas",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          swapOpacity={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-circle fa-w-16 "
                            color="#8B8B90"
                            data-icon="circle"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 512 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                        <styled__StepIndex
                          className="fa-layers-text fa-inverse"
                        >
                          <StyledComponent
                            className="fa-layers-text fa-inverse"
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "styled__StepIndex-g8vnjr-2",
                                  "isStatic": false,
                                  "lastClassName": "c3",
                                  "rules": Array [
                                    "font-size:14px;font-weight:bold;",
                                  ],
                                },
                                "displayName": "styled__StepIndex",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "styled__StepIndex-g8vnjr-2",
                                "target": "span",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <span
                              className="c3 fa-layers-text fa-inverse"
                            >
                              2
                            </span>
                          </StyledComponent>
                        </styled__StepIndex>
                      </span>
                    </StyledComponent>
                  </styled__IconLayer>
                </StepIcon>
                <styled__StepLabel
                  active={false}
                >
                  <StyledComponent
                    active={false}
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "styled__StepLabel-g8vnjr-3",
                          "isStatic": false,
                          "lastClassName": "c4",
                          "rules": Array [
                            "margin-left:8px;font-weight:",
                            [Function],
                            ";",
                          ],
                        },
                        "displayName": "styled__StepLabel",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "styled__StepLabel-g8vnjr-3",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="c4"
                    >
                      Step 2
                    </span>
                  </StyledComponent>
                </styled__StepLabel>
              </div>
            </StyledComponent>
          </styled__StepWrapper>
        </Step>
        <styled__StepDivider
          key=".2/.0"
          vertical={false}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styled__StepDivider-g8vnjr-1",
                  "isStatic": false,
                  "lastClassName": "c5",
                  "rules": Array [
                    "flex:1 1 auto;",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "styled__StepDivider",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styled__StepDivider-g8vnjr-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            vertical={false}
          >
            <div
              className="c5"
            />
          </StyledComponent>
        </styled__StepDivider>
        <Step
          active={false}
          complete={false}
          disabled={false}
          index={3}
          key=".2/.1"
        >
          <styled__StepWrapper>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styled__StepWrapper-g8vnjr-4",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "display:flex;align-items:center;padding:10px;cursor:",
                      [Function],
                      ";",
                    ],
                  },
                  "displayName": "styled__StepWrapper",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styled__StepWrapper-g8vnjr-4",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                <StepIcon
                  active={false}
                  complete={false}
                  index={3}
                >
                  <styled__IconLayer
                    className="fa-layers"
                  >
                    <StyledComponent
                      className="fa-layers"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "styled__IconLayer-g8vnjr-0",
                            "isStatic": false,
                            "lastClassName": "c2",
                            "rules": Array [
                              "font-size:24px;min-width:24px;",
                            ],
                          },
                          "displayName": "styled__IconLayer",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "styled__IconLayer-g8vnjr-0",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <span
                        className="c2 fa-layers"
                      >
                        <FontAwesomeIcon
                          border={false}
                          className=""
                          color="#8B8B90"
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                512,
                                512,
                                Array [],
                                "f111",
                                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z",
                              ],
                              "iconName": "circle",
                              "prefix": "fas",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          swapOpacity={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-circle fa-w-16 "
                            color="#8B8B90"
                            data-icon="circle"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 512 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                        <styled__StepIndex
                          className="fa-layers-text fa-inverse"
                        >
                          <StyledComponent
                            className="fa-layers-text fa-inverse"
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "styled__StepIndex-g8vnjr-2",
                                  "isStatic": false,
                                  "lastClassName": "c3",
                                  "rules": Array [
                                    "font-size:14px;font-weight:bold;",
                                  ],
                                },
                                "displayName": "styled__StepIndex",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "styled__StepIndex-g8vnjr-2",
                                "target": "span",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <span
                              className="c3 fa-layers-text fa-inverse"
                            >
                              3
                            </span>
                          </StyledComponent>
                        </styled__StepIndex>
                      </span>
                    </StyledComponent>
                  </styled__IconLayer>
                </StepIcon>
                <styled__StepLabel
                  active={false}
                >
                  <StyledComponent
                    active={false}
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "styled__StepLabel-g8vnjr-3",
                          "isStatic": false,
                          "lastClassName": "c4",
                          "rules": Array [
                            "margin-left:8px;font-weight:",
                            [Function],
                            ";",
                          ],
                        },
                        "displayName": "styled__StepLabel",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "styled__StepLabel-g8vnjr-3",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="c4"
                    >
                      Step 3
                    </span>
                  </StyledComponent>
                </styled__StepLabel>
              </div>
            </StyledComponent>
          </styled__StepWrapper>
        </Step>
      </div>
    </StyledComponent>
  </styled__StepperWrapper>
</Stepper>
`;
